image/image.go

```diff
@@ -115,6 +122,10 @@ func Unpack(tarFile, dest, ref string) error {
 }
 
 func unpack(w walker, dest, refName string) error {
+	if err := refsNameRegexpCheck(refName); err != nil {
```

Comment:
On Sun, Oct 30, 2016 at 07:11:09PM -0700, TrumanLing wrote:
> … but we can't avoid user's typing error…

That's not a problem.  If the user typo's the ref name, unpacking will fail because the ref does not exist.  Checking against the regexp in that case only “protects” you from the unlikely and presumably intentional case where the user is trying to unpack an illegal ref *that already exists in the ref store*.  If the illegal ref is in the ref store it's not a compliant image, but I still think we should allow the user to unpack the ref (assuming the DAG it points to is unpackable).